buildenv/jenkins/common/variables-functions.groovy

```diff
@@ -702,19 +702,151 @@ def set_slack_channel() {
+            //println testNodes
+            for (node in testNodes) {
+                def nodeGeo = get_node_geo(node.getLabelString())
+                def nodePlatform = get_node_platform(node.getLabelString())
+                //println nodeGeo
+                //println nodePlatform
+                // Upload if there is a server at geo where there are machines matching our platform.
+                if (nodePlatform == compilePlatform && ARTIFACTORY_CONFIG['geos'].contains(nodeGeo)) {
+                    ARTIFACTORY_CONFIG[nodeGeo]['uploadBool'] = true
+                    //println ARTIFACTORY_CONFIG[nodeGeo]['uploadBool']
+                }
+            }
+        }
+
+        echo "ARTIFACTORY_CONFIG:'${ARTIFACTORY_CONFIG}'"
+        /*
+        * Write out default server values to string variables.
+        * The upstream job calls job.getBuildVariables() which only returns strings.
+        * Rather than parsing out the ARTIFACTORY_CONFIG map that is stored as a string,
+        * we'll write out the values to env here as strings to save work later.
+        */
+        env.ARTIFACTORY_SERVER = ARTIFACTORY_CONFIG[ARTIFACTORY_CONFIG['defaultGeo']]['server']
+        env.ARTIFACTORY_REPO = ARTIFACTORY_CONFIG['repo']
+        env.ARTIFACTORY_NUM_ARTIFACTS = ARTIFACTORY_CONFIG[ARTIFACTORY_CONFIG['defaultGeo']]['numArtifacts']
+        env.ARTIFACTORY_DAYS_TO_KEEP_ARTIFACTS = ARTIFACTORY_CONFIG[ARTIFACTORY_CONFIG['defaultGeo']]['daysToKeepArtifacts']
+        env.ARTIFACTORY_MANUAL_CLEANUP = ARTIFACTORY_CONFIG[ARTIFACTORY_CONFIG['defaultGeo']]['manualCleanup']
+    }
+}
+
+def get_node_geo(nodeLabels) {
+    if (nodeLabels.contains('ci.geo.')) {
+        labelArray = nodeLabels.tokenize()
+        for (label in labelArray) {
+            if (label ==~ /ci\.geo\..*/) {
+                return label.substring(7)
+            }
+        }
     }
+    return ''
+}
+
+def get_node_platform(nodeLabels) {
+    /*
+    * xlinux -> arch=x86 && baseOS=linux
+    * plinux -> arch=ppc64le && baseOS=linux
+    * zlinux -> arch=s390x && baseOS=linux
+    * aix -> baseOS=aix
+    * windows -> baseOS=windows
+    * osx -> baseOS=osx
+    * zos -> baseOS=zos
+    * aarch64 -> arch=aarch64
+    */
+    def arch = ''
+    def baseOS = ''
+    if (nodeLabels.contains('hw.arch.') && nodeLabels.contains('sw.os.')) {
+        labelArray = nodeLabels.tokenize()
+        for (label in labelArray) {
+            switch(label) {
+                case ~/hw\.arch\.[a-z0-9]+/:
+                    arch = label.substring(8)
+                    //println arch
+                    break
+                case ~/sw\.os\.[a-z]+/:
+                    baseOS = label.substring(6)
+                    //println baseOS
+                    break
+            }
+        }
+    }
+    if (!arch || !baseOS) {
+        echo "WARNING: Unable to determine node arch/os:'${nodeLabels}'"
+        return ''
+    }
+    switch(baseOS) {
+        case 'aix':
```

Comment:
Could you please simplify the switch statement below, maybe use an array for all the matching cases sharing the same code, for example:
```
def nodePlatform
switch(baseOS) {
    case ['aix, 'windows',  'osx', 'zos'] :
        nodePlatform  = baseOS
        break
    case ['linux', 'ubuntu', 'rhel', 'cent', 'sles']:
        ...
        break
    default:
        echo "WARNING: Unknown baseOS:'${baseOS}'"
        nodePlatform  = ''
        break
    }
return nodePlatform 
```